panel/src/panel/content.js

```diff
@@ -145,17 +147,65 @@ export default (panel) => {
 			try {
 				await panel.api.post(api + "/changes/publish", values);
 
+				// close the retry dialog if it is still open
+				this.dialog?.close();
+
 				// update the props for the current view
 				if (this.isCurrent(api)) {
 					panel.view.props.originals = panel.view.props.content;
 				}
 
 				panel.events.emit("content.publish", { values, api });
+			} catch (error) {
+				this.retry("publish", error, panel.view.props.content, api);
 			} finally {
 				this.isProcessing = false;
 			}
 		},
 
+		/**
+		 * Opens a dialog with the error message
+		 * to retry the given method.
+		 */
+		retry(method, error, values, api) {
+			// log the error to the console to make it
+			// easier to debug the issue
+			console.error(error);
+
+			// set the dialog instance
+			this.dialog = panel.dialog;
+
+			// show a dialog to the user to try again
+			this.dialog.open({
+				component: "k-text-dialog",
+				props: {
+					text: panel.t(`form.${method}.error`),
+					cancelButton: panel.t("close"),
+					submitButton: {
+						icon: "refresh",
+						text: panel.t("retry")
+					}
+				},
+				on: {
+					close: () => {
+						this.dialog = null;
+					},
+					submit: async () => {
+						this.dialog.isLoading = true;
+
+						// try again with the latest state in the props
+						await this.save(panel.view.props.content, api);
```

Comment:
```suggestion
						await this[method](panel.view.props.content, api);
```

Otherwise when retrying publish, it's calling save... and I don't think it should or am I missing something?

Reply:
Good catch! I just fixed it.